metrics/sink.go

```diff
@@ -36,11 +35,9 @@ func (c *CounterSink) Add(s Sample) {
 // IsEmpty indicates whether the CounterSink is empty.
 func (c *CounterSink) IsEmpty() bool { return c.First.IsZero() }
 
-func (c *CounterSink) Format(t time.Duration) map[string]float64 {
-	return map[string]float64{
-		"count": c.Value,
-		"rate":  c.Value / (float64(t) / float64(time.Second)),
-	}
+// Rate computes the rate per second of the aggregated values.
+func (c *CounterSink) Rate(t time.Duration) float64 {
+	return c.Value / (float64(t) / float64(time.Second))
```

Comment:
Should we be careful with division by zero here? If `t == 0`, you get `c.Value / (0 / time.Second)`, we'll get a `panic: runtime error: integer divide by zero`. We could check for `t == 0`, and if that's the case, return `t`?

I know that we check it one level higher (I think I did introduce that after it introduced a bug in a previous version). But seeing it isolated now, I do think it would be worth having that check locally too.

 What do you think?